includes/class-amp-theme-support.php

```diff
@@ -324,10 +324,30 @@ public static function handle_xhr_request() {
 
 		if ( isset( $pagenow ) && 'wp-comments-post.php' === $pagenow ) {
 			// We don't need any data, so just send a success.
-			add_filter( 'comment_post_redirect', function() {
+			add_filter( 'comment_post_redirect', function( $url, $comment ) {
+				// Get theme support.
+				$theme_support = get_theme_support( 'amp' );
+
+				// Add the comment ID to the URL to force AMP to refresh the page.
+				$url = add_query_arg( 'comment', $comment->comment_ID, $url );
+
+				// Send redirect header if amp-live-list has been opted-out.
+				if ( empty( $theme_support['comments_live_list'] ) ) {
+					header( 'AMP-Redirect-To: ' . $url, true );
+				}
+				// Create a success message to display to the user.
+				if ( '1' === (string) $comment->comment_approved ) {
+					$message = __( 'Your comment has been posted and has been approved.', 'amp' );
+				} else {
+					$message = __( 'Your comment has been posted, and is awaiting moderation.', 'default' );
```

Comment:
This string has the `default` text domain because it is re-using a string in core:

https://github.com/WordPress/wordpress-develop/blob/8f95800d52c1736d651ae6e259f90ad4a0db2c3f/src/wp-includes/class-walker-comment.php#L285

We need it to keep it the same for it to re-use core translations.